types/interval_compound_type.go

```diff
@@ -71,3 +71,7 @@ func (m IntervalCompoundType) String() string {
 	return fmt.Sprintf("intervalcompound%s<%d>", strNullable(m),
 		m.precision.ToProtoVal())
 }
+
+func (m IntervalCompoundType) Match(ot Type) bool {
+	return m == ot
```

Comment:
delegate to `m.Equals(ot)`?

Reply:
not needed since API changed to "MatchWithNullability" and "MatchWithoutNullability"